### DISCUSSION TITLE
Execution blocked by ES5 and require() error messages

### DISCUSSION THREAD

**Discussion by ASQABenny** [webdriverio/webdriverio#9812] [Q&A]
Hey guys,

in a test automation with the WebdriverIO framework using JS, we are currently having issues with the project continuing to run.
The project runs test automation for the desktop Windows environment in the Chrome browser, an Android app and an iOS app. For Android and iOS, the Appium service is also used and the iOS is run on a macOS.

After a large merge of 3 branches in GitHub to the Main Branch, issues have now arisen and we are looking for solutions to them to no avail.The desktop environment is working again after some fixes, but the Android and iOS environment is not.
The following errors occur as soon as we want to run the test automation:

**The use of export is no longer accepted.**
Error message: unexpected token 'export'.
This is needed for app-paths, for example.
Code snippet: `export const iosAppPath = projectPath.join(process.cwd(), "/app/appname.ipa")`

The error message suggests to use "ES5" by adding "type = 'modules'" in the package.json. We tried this and get the error message that imports with require() are no longer accepted.

Can anyone help us how to fix the problem easily?

**Comment by christian-bromann** [comment]
> The desktop environment is working again after some fixes, but the Android and iOS environment is not.

This error doesn't seem like it is based on the capabilities you run but more your set up.

> Error message: unexpected token 'export'.

This means you are using ESM logic while your project runs in a CJS environment.

> We tried this and get the error message that imports with require() are no longer accepted.

Switching to ESM completely might require some more changes to your codebase, e.g. `__dirname` is not available anymore and needs to be defined by you.

I recommend to read up on ESM vs CJS and ensure your code base is set up correctly. That is all I can say based on the information provided.
